### PR TITLE
Mark Aspire.Hosting.Python as experimental

### LINKED ISSUE #6298: Mark AddPythonApp API as experimental
We have a couple of issues with the existing AddPythonApp API. We want to mark it as experimental because it still needs more work.

### PR DESCRIPTION
**Pull request by sebastienros**
## Description

Fixes #6298

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [ ] Yes
  - [x] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Link to aspire-docs issue: 
  - [x] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6340)

### CODE REVIEW

**Comment by davidfowl** [comment]
We want to mark the API as experimental. When you call AddPythonApp it should have the attribute on it. Like this https://github.com/eerhardt/aspire/blob/5603e928797a629d972c0084e2d6414169d1295c/src/Aspire.Hosting.Azure.CognitiveServices/AzureOpenAIExtensions.cs#L39

**Comment by davidfowl** [comment]
Although, maybe pre-release is enough to signal what we want.

**Comment by sebastienros** [comment]
I could not find this attribute used anywhere though.

**Review by davidfowl — APPROVED** [review]

File: src/Aspire.Hosting.Python/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<Project>
+    <PropertyGroup>
+      <!-- This needs to be set to true before importing parent Directory.Build.props -->
+      <!-- In preview until the public API is validated. -->
+      <SuppressFinalPackageVersion>true</SuppressFinalPackageVersion>
```
Comment (davidfowl) [inline src/Aspire.Hosting.Python/Directory.Build.props:5]: Why isn't this in the csproj?

**Comment by eerhardt** [comment]
My thoughts:

1. Making a nuget package not stable (i.e. have a `-preview` suffix in the version) should be enough signal to people that the API can change before it becomes stable. Users don't have to install a "pre release" package and then ALSO suppress a warning about using experimental APIs inside that prerelease package.

2. `Aspire.Hosting.Python` already shipped stable in 8.1 and 8.2. If we don't ship a "stable" 9.0.0 package, people will continue using the 8.2.1 Aspire.Hosting.Python package with Aspire 9.0. This maybe won't cause problems, but it will be confusing.

File: src/Aspire.Hosting.Python/Directory.Build.props (hunk 1 above)
Comment (eerhardt) [reply src/Aspire.Hosting.Python/Directory.Build.props:5]: See conversation here: https://github.com/dotnet/aspire/pull/6195#discussion_r1792442697

cc @joperezr

Comment (sebastienros) [reply src/Aspire.Hosting.Python/Directory.Build.props:5]: Based on this PR this is the way to do it now
https://github.com/dotnet/aspire/pull/6195/files

**Comment by davidfowl** [comment]
@DamianEdwards thoughts?

File: src/Aspire.Hosting.Python/Directory.Build.props (hunk 1 above)
Comment (joperezr) [reply src/Aspire.Hosting.Python/Directory.Build.props:5]: That is no longer the case, it was only for RC1 as we had things going RC1 and some other things staying in preview, but now everything should fall back to be preview or stable.

**Comment by eerhardt** [comment]
Another problem is that shipped https://www.nuget.org/packages/Aspire.Hosting.Python/9.0.0-rc.1.24511.1. Which means when we ship `9.0.0` GA, the Python package produced with that build will still have `-preview` and will be LOWER than the rc1 package. So even if people move to the 9.0.* package, they will get the rc1 package, which is old.

Maybe we need to ship this as stable and mark the APIs as `[Experimental]`....

**Comment by DamianEdwards** [comment]
> Maybe we need to ship this as stable and mark the APIs as [Experimental]....

I thought that was the proposal?

**Comment by eerhardt** [comment]
> > Maybe we need to ship this as stable and mark the APIs as [Experimental]....
> 
> I thought that was the proposal?

What ended up getting changed was the package is no longer building as "stable", so when it gets built it will have a `-preview` version.

**Comment by sebastienros** [comment]
I will do the change

**Pull request merged**